pkg/repo/index_test.go

```diff
@@ -0,0 +1,55 @@
+package repo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	testRepo = "test-repo"
+)
+
+func TestDownloadIndexFile(t *testing.T) {
+	fileBytes, err := ioutil.ReadFile("testdata/local-index.yaml")
+	if err != nil {
+		t.Errorf("%#v", err)
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "binary/octet-stream")
+		fmt.Fprintln(w, string(fileBytes))
+	}))
+
+	path := filepath.Join("testdata", testRepo+"-index.yaml")
```

Comment:
If this is the destination file, you might consider using `ioutil.TempDir()` and saving it there. Not a PR blocker, though.